R/ExpandingMplusItemNames.R

```diff
@@ -0,0 +1,194 @@
+###     test variable names           ###
+### test0     F1-F10...               ###  Pass
+### test1     F1_1-F1_10...           ###  Pass
+### test2     F1_1_1-F1_10...         ###  Pass
+### test3     F1_1-F1_7,F26,F27,F28...###  Pass
+### test4     Fa-Fj...                ###  Pass
+### test5     F_a-F_j...              ###  Pass
+### test6     F_1_a-F_1_j...          ###  Pass
+### test7     Faa-Faj                 ###  Pass
+
+
+
+
+Groups <- unique(fit$parameters$unstandardized$Group)
+### Read dataset in the targetfile
+TargetFile <- {
+  if (grepl("\\.dat$",fit$input$data$file) == T){
+    read.table(fit$input$data$file)
+  }
+  else {
+    read.csv(fit$input$data$file,header = F, sep = ",")
+  }
+}
+
+## extract mplus variable names 
+## need test more variable names...
+ExpandColname <-{
+  MplusVarNames = toupper(fit$input$variable$names)
+  
+  #### split Mplusvarnames into vector sep = ", or space"
+  
+  MplusVarNamescomma = unlist(strsplit(MplusVarNames,",|\\s+"))
+  #### split Mplusvarnames into vector furthe sep = "-"
+  
+  MplusVarNameshyphen = strsplit(MplusVarNamescomma,"-")
+  
+  #### extract variable numbers in mplus, eg. u1-u6 or ua - uf returns c(1,6) or c("a","f")
+  ExtractedNumber <-{
+    PreExtractedNumberList = list()
+    ExtractedNumberList = list()
+    for (i in 1:length(MplusVarNamescomma)){
+      PreExtractedNumberList[[i]] =  stringr::stri_sub(MplusVarNameshyphen[[i]],-1,-1) 

```

Comment:
I believe this function is in `stringi`, not `stringr`
Also note that you only get the last digit of multi-digit numbers

Reply:
Sorry for that typo...
I saw that part. PreExtractedNumberList returns an array of the last digits, which help me find whether the items name includes only numbers or include numbers and letters (see line 45). According to Mplus naming rule, item names like u1-u6, ua-uj... can exist together. It may have a better way to rewrite this...Thank you for reviewing again.